be/src/exprs/vectorized/encryption_functions.cpp

```diff
@@ -199,4 +200,137 @@ ColumnPtr EncryptionFunctions::md5(FunctionContext* ctx, const Columns& columns)
     return result.build(ColumnHelper::is_all_const(columns));
 }
 
+Status EncryptionFunctions::sha2_prepare(starrocks_udf::FunctionContext* context,
+                                         starrocks_udf::FunctionContext::FunctionStateScope scope) {
+    if (scope != FunctionContext::FRAGMENT_LOCAL) {
+        return Status::OK();
+    }
+
+    ColumnPtr column = context->get_constant_column(1);
```

Comment:
First you should check if the second column is const.
If it is const, this optimization can be leveraged.